src/descriptor/mod.rs

```diff
@@ -416,6 +419,37 @@ impl<P: ToPublicKey> Descriptor<P> {
     }
 }
 
+impl<P: ToPublicKey + Clone> Descriptor<P> {
+    pub fn interpret(
+        &self,
+        secp: &secp256k1::Secp256k1<VerifyOnly>,
+        stack: &mut Vec<StackElement>,
+        sighash: &secp256k1::Message,
+        age: u32,
+    ) -> Result<Primitives, Error>
+    {
+        match *self {
+            Descriptor::Bare(ref d) |
+            Descriptor::Sh(ref d) |
+            Descriptor::ShWsh(ref d) |
+            Descriptor::Wsh(ref d) => {
+                d.interpret(secp, stack, sighash, age)
+            },
+            Descriptor::Pkh(ref pk) |
+            Descriptor::ShWpkh(ref pk) |
+            Descriptor::Wpkh(ref pk) => {
+                let sig = stack.pop().expect("Signature").is_witness_or_err()?;
```

Comment:
This `expect` is invalid because there might not be anything on the stack when the user calls this function. See my below comment about `is_witness_or_err` - I think you should give the `Option<_>` from `pop` directly to that method.